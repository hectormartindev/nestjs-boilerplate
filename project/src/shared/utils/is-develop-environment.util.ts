import AppConfig from '../../app.config'

const isDevelopEnvironment = (): boolean => {
  return AppConfig.deployEnvironment === 'dev'
}

export default isDevelopEnvironment
