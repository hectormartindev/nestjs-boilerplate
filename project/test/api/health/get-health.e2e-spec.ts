import { HttpStatus } from '@nestjs/common'
import DatabaseHealthIndicatorFake from '../../../src/shared/infrastructure/database/health-indicator.provider.fake'
import { createClient } from '../../utils/test-client'

const HEALTH_URL = '/health'

// TODO: check for bodys
describe(`${HEALTH_URL} (GET)`, () => {
  it('returns ok health status', async () => {
    const client = await createClient()

    const { body } = await client.health().expect(HttpStatus.OK).run()
    expect(body).toMatchSnapshot()
  })

  it('returns ko health status', async () => {
    const client = await createClient({
      databaseHealthIndicator: new DatabaseHealthIndicatorFake(true),
    })

    const { body } = await client.health().expect(HttpStatus.SERVICE_UNAVAILABLE).run()
    expect(body).toMatchSnapshot()
  })
})
