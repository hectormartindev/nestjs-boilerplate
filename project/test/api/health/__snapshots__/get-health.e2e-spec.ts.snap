// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/health (GET) returns ko health status 1`] = `
Object {
  "details": Object {
    "down": Object {
      "status": "down",
    },
  },
  "error": Object {
    "down": Object {
      "status": "down",
    },
  },
  "info": Object {},
  "status": "error",
}
`;

exports[`/health (GET) returns ok health status 1`] = `
Object {
  "details": Object {
    "up": Object {
      "status": "up",
    },
  },
  "error": Object {},
  "info": Object {
    "up": Object {
      "status": "up",
    },
  },
  "status": "ok",
}
`;
